sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -1488,6 +1490,20 @@ private Result(PCollectionTuple pct) {
        * @return the resources
        */
       public PCollection<JsonArray> getResources() {
+        return resources
+            .apply(
+                "Extract Values",
+                MapElements.into(TypeDescriptor.of(JsonArray.class))
+                    .via((KV<String, JsonArray> in) -> in.getValue()))
+            .setCoder(JsonArrayCoder.of());
+      }
```

Comment:
If a user calls this function many times (to use the resources collection in multiple places), we would append multiple `MapElements` transforms to the pipeline - this may not be the effect they would expect. Perhaps checking to only create unkeyed resources once?

Reply:
Yeah, I was debating if it was better to have 2 PCollections or to have multiple transforms.

If we're going to store both keyed (PCollection<KV<String, JsonArray>>) & unkeyed (PCollection<JsonArray>) resources in the Search.Result, is it better to just create both in the constructor? Or only populate unkeyed if getResources() has been called?

I believe the first would keep the MapElements step as part of the FhirIO.Search operation, but the second would still add a (single?) extra MapElements to a user's pipeline.

Reply:
oh you're right. Good thought.... With that in mind, I am inclined to just add both to the Search.Result object when creating it. 

The shortcoming is that we will always running the `MapElements` to discard the key; but it should be a cheap operation. The benefit is that it will not confuse users in the UI. WDYT?

Reply:
I agree, adding both to the object is the best option & definitely makes it the easiest to read.